lib/Data/ValueProxy.js

```diff
@@ -286,7 +287,14 @@ export class ValueProxy extends Events {
      * @returns {ValueProxy}
      */
     static all(proxies) {
+        let generation = 0;
+
+        proxies.forEach(proxy => {
```

Comment:
You can replace `forEach` and top scope variable mutation to simple `reduce` method

```
let generation = proxies.reduce((generation, proxy) => {
    return Math.max(generation, proxy._generation + 1)
}, 0);
```